### ISSUE TITLE
Once any included devices have been added, it is impossible to not specify included devices

### ISSUE THREAD

**Issue by JasonBSteele** [alandtse/alexa_media_player#2751]
**IMPORTANT: Please search the issues, including closed issues, and the [FAQ](https://github.com/alandtse/alexa_media_player/wiki/FAQ) before opening a new issue. The template is mandatory; failure to use it will result in issue closure.**

**Describe the bug**
If you enter any devices in the "Included devices" field of the Reconfiguration dialog and save you cannot then re-edit "included devices" to be blank. An empty string is not saved. A space is saved but still means that only included devices will be included.

**To Reproduce**
1. Click CONFIGURE and enter a device in the "included devices" field and save 
2. Click CONFIGURE again and delete the device you entered leaving an empty string and save
3. Click CONFIGURE again and you will see the device hasn't been removed

**Expected behavior**
I would expect the device entered to be removed when I delete the text leaving an empty string and save,

**System details**

- Home Assistant version:
Core 2024.12.3
Supervisor 2024.11.4
Operating System 14.0
Frontend 20241127.8

- alexa_media version (from `const.py` or HA startup log): 5.0.4

**Comment by danielbrunt57** [comment]
Can you try adding this code to `config_flow.py`?

```
            if CONF_INCLUDE_DEVICES in self.config_entry.data:
                if user_input[CONF_INCLUDE_DEVICES] == " ":
                    user_input[CONF_INCLUDE_DEVICES] = ""

            if CONF_EXCLUDE_DEVICES in self.config_entry.data:
                if user_input[CONF_EXCLUDE_DEVICES] == " ":
                    user_input[CONF_EXCLUDE_DEVICES] = ""
```

![image](https://github.com/user-attachments/assets/c4d7d008-7611-4da0-81c2-fec085ceb4c2)

**Comment by JasonBSteele** [comment]
I have already resolved the issue for for me by editing the attribute "include_devices" in config.storage\core.config_entries so that it is an empty string.

I can give your fix a go for testing purposes (i.e. break it again, then fix it with this code) but it is probably straight forward enough not to need that.

**Comment by M0rn1ngst4rr** [comment]
tryed Both ways so added the included code and changed the config_entries but it won't find any devices anymore.

every time i reload the string in include_devices and exclude_devices is " " again.

**Comment by danielbrunt57** [comment]
Simple workaround is to uninstall the integration and reinstall it.

**Comment by danielbrunt57** [comment]
> Also, I wonder if strip() might be useful. That way it wouldn't matter how many spaces had been entered, and it might be useful anyway to remove unnecessary whitespaces from the ends of the string of devices

Yes!
```
>>> txt = "  This device,Next Device,Ths is the last device  "
>>> print("\"" + txt.strip() + "\"")
"This device,Next Device,Ths is the last device"
>>>
```
```
>>> txt = " "
>>> print("\"" + txt + "\"")
" "
>>> print("\"" + txt.strip() + "\"")
""
>>>
```

**Comment by danielbrunt57** [comment]
Replace
```
        if user_input is not None:
            """Preserve these parameters"""
            if CONF_URL in self._config_entry.data:
                user_input[CONF_URL] = self._config_entry.data[CONF_URL]
            if CONF_EMAIL in self._config_entry.data:
                user_input[CONF_EMAIL] = self._config_entry.data[CONF_EMAIL]
            if CONF_PASSWORD in self._config_entry.data:
                user_input[CONF_PASSWORD] = self._config_entry.data[CONF_PASSWORD]
            if CONF_SECURITYCODE in self._config_entry.data:
                user_input[CONF_SECURITYCODE] = self._config_entry.data[
                    CONF_SECURITYCODE
                ]
            if CONF_OTPSECRET in self._config_entry.data:
                user_input[CONF_OTPSECRET] = self._config_entry.data[CONF_OTPSECRET]
            if CONF_OAUTH in self._config_entry.data:
                user_input[CONF_OAUTH] = self._config_entry.data[CONF_OAUTH]
            """Ensure public_url ends with trailing slash"""
            if CONF_PUBLIC_URL in self._config_entry.data:
                if not user_input[CONF_PUBLIC_URL].endswith("/"):
                    user_input[CONF_PUBLIC_URL] = user_input[CONF_PUBLIC_URL] + "/"

            self.hass.config_entries.async_update_entry(
                self._config_entry, data=user_input, options=self._config_entry.options
            )
            return self.async_create_entry(title="", data={})
 ```
 With
 ```
        if user_input is not None:
            """Preserve these parameters"""
            if CONF_URL in self._config_entry.data:
                user_input[CONF_URL] = self._config_entry.data[CONF_URL]
            if CONF_EMAIL in self._config_entry.data:
                user_input[CONF_EMAIL] = self._config_entry.data[CONF_EMAIL]
            if CONF_PASSWORD in self._config_entry.data:
                user_input[CONF_PASSWORD] = self._config_entry.data[CONF_PASSWORD]
            if CONF_SECURITYCODE in self._config_entry.data:
                user_input[CONF_SECURITYCODE] = self._config_entry.data[
                    CONF_SECURITYCODE
                ]
            if CONF_OTPSECRET in self._config_entry.data:
                user_input[CONF_OTPSECRET] = self._config_entry.data[CONF_OTPSECRET]
            if CONF_OAUTH in self._config_entry.data:
                user_input[CONF_OAUTH] = self._config_entry.data[CONF_OAUTH]
            """Ensure public_url ends with trailing slash"""
            if CONF_PUBLIC_URL in self._config_entry.data:
                if not user_input[CONF_PUBLIC_URL].endswith("/"):
                    user_input[CONF_PUBLIC_URL] = user_input[CONF_PUBLIC_URL] + "/"
            if CONF_INCLUDE_DEVICES in self._config_entry.data:
                user_input[CONF_INCLUDE_DEVICES] = user_input[CONF_INCLUDE_DEVICES].strip()
            if CONF_EXCLUDE_DEVICES in self._config_entry.data:
                user_input[CONF_EXCLUDE_DEVICES] = user_input[CONF_EXCLUDE_DEVICES].strip()
            self.hass.config_entries.async_update_entry(
                self._config_entry, data=user_input, options=self._config_entry.options
            )
            return self.async_create_entry(title="", data={})
```

**Comment by danielbrunt57** [comment]
> If you enter any devices in the "Included devices" field of the Reconfiguration dialog and save you cannot then re-edit "included devices" to be blank. An empty string is not saved. A space is saved but still means that only included devices will be included.

My question to anyone using "Include devices" is Why are you using that field?  Are you trying to limit the resulting devices in AMP to less than 50% of the possible devices?  Because that would be the only reason.  If the "Include devices" list is more than 50%, use "Exclude Devices" as that would be a smaller list.

**Comment by JasonBSteele** [comment]
> My question to anyone using "Include devices" is Why are you using that field?
For me it was a misunderstanding of how it worked. I thought I needed to explicitly add new devices to the list. When I realized that wasn't how it worked, that's when I ran into problems trying to remove it.

Having said that, I have disabled a lot of entities that it is creating because unfortunately all the HA entities that the Alexa Integration has created in Alexa get recreated again!

So I probably do need to tidy this up by using Include or Exclude. And in fact Include would probably be better as there a lot to be excluded.
